### DISCUSSION TITLE
PDF annotations

### DISCUSSION THREAD

**Discussion by abulusky** [saber-notes/saber#1168] [Ideas]
Thank you for the fantastic app and great effort! I'm thoroughly enjoying using Saber.

One feature that I believe would significantly enhance the app's functionality is PDF annotation. We often need to go through various PDFs and annotating becomes a necessary feature. Sadly, there's a no decent open-source Android app, if any that supports annotation capabilities, making the feature a demanding one.

Given Saber's already toolsets like pen, pencil, highlighters, text, shape etc, incorporating lossless PDF import functionality along with necessary modifications for annotations would elevate its utility even further. It's the one missing piece that's preventing me from fully transitioning from other apps to Saber. I believe, provided the excellent existing toolsets and features of Saber, implementing the feature wont be very difficult.

Please consider the feature.

**Comment by axiomtutor** [comment]
I tested this app for PDF annotation.  There are a few missing features that prevent me from being able to use Saber for this.

1) No infinite canvas, only fixed-size pages.  I really need to be able to zoom/pan outside the PDF pages and put a lot of hand-written text around the page.  As far as I can tell, right now Saber doesn't support this at all.

2) No zoom ability.  I often want to be able to zoom into a part of the PDF in order to focus the viewer's attention on the part that I'm discussing.  

3) A lower priority, but which could still be nice, is the ability to draw text boxes, so that I can place typed text in certain regions.  (Even better if it supports some kind of TeX plugin like KaTeX or something!)

**Comment by jyo64** [comment]
I second the ability to draw text boxes. This would be very useful for me when taking lecture note points on top of PDFs.
